camel/messages/base.py

```diff
@@ -26,370 +26,306 @@
     OpenAIUserMessage,
 )
 from camel.prompts import CodePrompt, TextPrompt
-from camel.types import (
-    OpenAIBackendRole,
-    OpenAIImageType,
-    OpenAIVisionDetailType,
-    RoleType,
-)
+from camel.types import MessageType, OpenAIBackendRole, RoleType
 from camel.utils import Constants
 
 
-@dataclass
-class BaseMessage:
-    r"""Base class for message objects used in CAMEL chat system.
+class Sender(BaseModel):
+    id: str
+
+
+class Receiver(BaseModel):
+    id: str
+
 
-    Args:
-        role_name (str): The name of the user or assistant role.
-        role_type (RoleType): The type of role, either :obj:`RoleType.
-            ASSISTANT` or :obj:`RoleType.USER`.
-        meta_dict (Optional[Dict[str, str]]): Additional metadata dictionary
-            for the message.
-        content (str): The content of the message.
-        video_bytes (Optional[bytes]): Optional bytes of a video associated
-            with the message. Default is None.
-        image_list (Optional[List[Image.Image]]): Optional list of PIL Image
-            objects associated with the message. Default is None.
-        image_detail (Literal["auto", "low", "high"]): Detail level of the
-            images associated with the message. Default is "auto".
-        video_detail (Literal["auto", "low", "high"]): Detail level of the
-            videos associated with the message. Default is "low".
-    """
+class Content(BaseModel):
+    text: Optional[List[str]] = Field(default_factory=list)
+    image_url: Optional[List[str]] = Field(default_factory=list)
+    image_detail: str = Field(default="auto")
+    video_url: Optional[List[str]] = Field(default_factory=list)
+    video_detail: str = Field(default="low")
+    audio_url: Optional[List[str]] = Field(default_factory=list)
 
+
+class Envelope(BaseModel):
+    time_sent: str
+    time_received: str
+
+
+class BaseMessage(BaseModel):
```

Comment:
I understand this is a draft, but I would suggest that we still add the comments, since they are required and help people review the code better.